_posts/2019-06-04-loading-large-image-data-with-dask-array.md

```diff
@@ -67,160 +173,923 @@ pipeline.
+import matplotlib.pyplot as plt
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
+```
+
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-1.png"
+     width="60%">
+
+```python
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
 ```
 
-If we want to simplify this a bit, we can use a pre-rolled implementation that
-does this work for us like [dask-image's `imread`](
-https://dask-image.readthedocs.io/en/latest/dask_image.imread.html ). This
-combines the simplicity of an API like scikit-image's with the benefits of
-leveraging Dask to manage the larger data.
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-2.png"
+     width="60%">
 
 ```python
-from dask_image.imread import imread
-a = imread("mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif")
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
 ```
 
-If we want to handle loading some other kind of data, we can take our own
-loading function and combine it with `delayed`. This way we can handle
-operations like reading from a database or requesting data from the cloud.
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-3.png"
+     width="60%">
 
-Combining data loaded with Dask
--------------------------------
 
-Thus far we have only showed how one might load one image file. However we
-generally have many such files. At a first pass we might try to load up the
-collection of files in a list.
+### Investigate Filename Structure
+
+This are slices from only one chunk of a much larger aggregate image.
+Our main task in dealing with large image stacks is how to arrange the pieces
+back together.
+It is common to see a naming structure in the filenames. Each
+filename then may indicate a channel, time step, and spatial location with the
+`<i>` being some numeric values (possibly with units). Individual filenames may
+have more or less information and may notate it differently than we have.
+
+```
+mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif  # TODO: use filename structure from the AOLLSM example
+```
+
+In principle with NumPy we might allocate a giant array and then iteratively
+load images and place them into the giant array.
 
 ```python
-from dask_image.imread import imread
-l = [imread(fn) for fn in myfiles]
+full_array = np.empty((..., ..., ..., ..., ...), dtype=sample.dtype)
+
+for fn in filenames:
+    img = skimage.io.imread(fn)
+    index = get_location_from_filename(fn)
+    full_array[index, :, :, :] = img
 ```
 
-This is fine, but it's not that easy to manage. It would be better if we could
-turn this into one Dask Array instead of a list of many. We have a couple
-options. If we want to combine our Dask Arrays along an *existing* axis, we can
-use [`concatenate`](
-http://docs.dask.org/en/latest/array-api.html#dask.array.concatenate ).
+But in practice large image data is often too large to fit into memory, so we
+use Dask.
```

Comment:
I think if we present it this way, someone is going to ask about memory-mapping. Should we add that?